### ISSUE TITLE
products not showing on frontend after migration

### ISSUE THREAD

**Issue by CNemo01** [magento/data-migration-tool#572]
### Preconditions

1. Magento 1.8.1, PHP 5.6.31, MySQL 5.5
2. Magento 2.2.5, PHP 7.0.3, MariaDB 10.1.33

After migrating a project from Magento 1.8.1 to Magento 2.2.5, the data was successfully transferred. However, although all products were listed in the backend and correctly linked to the appropriate website and shop, the products were not listed in the frontend (for users not logged in). After more than 1 day investigating and searching, I found a very simple cause:

In the table [customer_group] in our Magento 1.8.1 installation, the **_customer_group_id_** for the system generated group "**_NOT LOGGED IN_**" was set to "**_9_**". This id was transferred to the Magento 2.2.5 installation. When changing the id to "**_0_**", all products were shown in the frontend after flushing cache and reindexing.

### Steps to reproduce

1. Locate a Magento 1.x version with the customer group "NOT LOGGED IN" installed with a customer_group_id value anything else than "0"
2. Migrate the data to a fresh install of Magenot 2.2.x
3. flush cache and reindex, then check the shop without being logged in as customer

### Expected result

1. If the migration ends successfully, the frontend of the shop should show the products which were successfully migrated and displayed in the backend (assuming that inventory settings are correct)

### Actual result

1. Products are not shown in the frontend for non-logged-in users
2. The correct categories are displayed with product counts "0"
3. Products are correctly listed and linked to the shop in the backend
4. Also the magento search does not show any results

### Additional notes
The migration-tool should automatically set the customer_group_id for the group "NOT LOGGED IN" appropriately according to the magento code. Alternatively, this can be an interesting easy way to implement a solution to magento for displaying products only to logged customers...

**Comment by victor-v-rad** [comment]
Thank you @CNemo01 for the informative issue report. I checked all my instances of M1 and all customer_group_id have **0** for NOT LOGGED IN record. Could you check that on Edit Customer Group page for "NOT LOGGED IN" **is not editable**  in your Magento 1 admin panel? I also see in M1 \Mage_Customer_Model_Group::NOT_LOGGED_IN_ID is 0 and the same for M2. I suppose in your case some extension of Magento 1 may intervene. 
I am no sure that the tool should search for NOT LOGGED IN record and make it **0** if it is 9, for example. Because all the records from other tables may refer to 9 and not 0; What is possible to do is to inform developer during migration that no NOT_LOGGED_IN_ID with 0 in customer_group_id is found. Internal ticket MAGETWO-94116 to fix it

**Comment by CNemo01** [comment]
Hello @victor-v-rad, I can edit the customer group "NOT LOGGED IN" in the backend of Magento 1.8.1 but only the tax class can be changed, the ID is hidden and the name is uneditable. Unfortunately, I am unaware about what might have happened since we were contracted to migrate data and move the existing shop to Magento 2. Subsequently we did not install the Magento 1 version or the extensions (which are limited to a few for shipping and payment) and I can therefore not exclude any earlier faulty manipulations of the data by extensions or the former admin.

However, after I have tried several migrations of the data with the same effect of products not showing up in the frontend, I searched a lot and found one post that led me to the finding which relates to a possible earlier bug when the id was set to 4?! ->see here: [http://blog.hiremagentodeveloper.com/ecommerce/no-products-are-showing-on-the-front-end-after-magento-version-upgrade.html](url).

In the end: You might be right not forcing an automated change to 0 (although the system seems to relate on the group having the id 0 and personally I would rather take the risk to have an extension not working anymore due to a faulty group id rather than the system not working properly).
